### DISCUSSION TITLE
Default credentials being logged

### DISCUSSION THREAD

**Discussion by ariasdiana** [docker-library/rabbitmq#525] [Q&A] [answered]
I'm currently running RabbitMQ through docker-compose. I've noticed two things.

First one is that when I run "docker-compose up", upon initializing itself, my RabbitMQ container will log my default credentials (I am passing them through env variables). How do I go about avoiding this?

Second thing is, when I include a command on my compose file to modify log levels, it gets ignored.

I'm using the 3.8.19-management image. I've tried the same thing with alpine and am still getting the same issue though. Any advice? 

Thanks

**Comment by tianon** [comment, marked as answer]
I'm guessing the logging you're referring to is the part following the warning?

```console
WARNING: 'docker-entrypoint.sh' generated/modified the RabbitMQ configuration file, which will no longer happen in 3.9 and later! (https://github.com/docker-library/rabbitmq/pull/424)

Generated end result, for reference:
------------------------------------
loopback_users.guest = false
listeners.tcp.default = 5672
default_pass = bar
default_user = foo
------------------------------------
...
```

To avoid this, I'd suggest mounting/providing a configuration file instead of using environment variables (per https://github.com/docker-library/rabbitmq/pull/424).

Alternatively, you could update to 3.9 in which https://github.com/rabbitmq/rabbitmq-server/pull/3299 implemented built-in support for `RABBITMQ_DEFAULT_USER`/`RABBITMQ_DEFAULT_PASS`.

**Reply by ariasdiana** [reply]
That did the trick. Thank you!
